### DISCUSSION TITLE
Test Java files are not processed

### DISCUSSION THREAD

**Discussion by spartanhooah** [diffplug/spotless#1573] [Q&A] [answered]
I'm using the Gradle plugin version 6.11.0. I have noticed that test files (in `src/test/java/..`) are not getting processed by Spotless. No header is added, unused imports aren't removed, etc. I can add "wrong" code to files in `src/main/java` and they get corrected just fine. Spotless is configured like this:
```
spotless {
    java { //tells spotless we are using java
        //defines the order that we want to import our dependencies
        importOrder 'java', 'javax', 'org', 'com', 'com.mycompany', ''
        //tells spotless to remove unused import statements that way nothing gets cluttered
        removeUnusedImports()
        //tells spotless to use the google java format more can be found at go/javastyleguide
        googleJavaFormat('1.15.0').aosp()
        licenseHeader '/** Copyright (C) $YEAR My Company All Rights Reserved */'
    }
}
```

How come test files aren't getting processed?

**Comment by jbduncan** [comment]
@spartanhooah Welcome! It's not immediately clear why this may be happening. Can you make a [minimal, reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) Gradle project and share it with us?

One idea that may work is to explicitly set the `target`, which is a glob expression for all your Java files. So assuming your Gradle project has a `src` directory at the same level as the Gradle file, you could try the following as per the [README](https://github.com/diffplug/spotless/tree/main/plugin-gradle#java):

```gradle
spotless {
    java {
        target 'src/*/java/**/*.java'
        importOrder 'java', 'javax', 'org', 'com', 'com.mycompany', ''
        removeUnusedImports()
        googleJavaFormat('1.15.0').aosp()
        licenseHeader '/** Copyright (C) $YEAR My Company All Rights Reserved */'
    }
}

```

**Reply by spartanhooah** [reply]
Adding this target glob did not help. I'll work on a minimum, reproducible example.

**Comment by jbduncan** [comment]
As a side note, you'll be glad to hear that you don't need `removeUnusedImports()`, because google-java-format does it too! (Spotless uses google-java-format to remove unused imports behind the scenes.)

**Comment by spartanhooah** [comment, marked as answer]
🤦 So the files that weren't getting processed were in a second project in IntelliJ. Of course they weren't getting fixed!

**Reply by jbduncan** [reply]
Oh, I see! I'm glad you figured out where things went wrong, then. I'll accept this as the answer, but if I've misunderstood or you need further help, then feel free to come back and I'll change things.
